### DISCUSSION TITLE
Tenants Disk Ressource Limitation

### DISCUSSION THREAD

**Discussion by tsohst** [minio/minio#15618] [Q&A] [answered]
**Is your feature request related to a problem? Please describe.**
We would like to setup a "Host Multiple Tenants on Multiple Drives (Erasure Code)"
https://docs.min.io/docs/multi-tenant-minio-deployment-guide.html

As the infrastructure below minio is shared over all tennants, we would like to limit a tennent (end-customer) to a limited amount of disk ressources available in the cluster.
This would help us having better ressource management and prevents a disk-capacity overload on the cluster, which would affect all customers as well.

**Describe the solution you'd like**
MINIO_ROOT user should be able to set a maximum disk limit on a tennant.

**Describe alternatives you've considered**
Quotas on a bucket might not help, as the end-customer can create himself as many buckets as he want.
Also end-customer can change quota on buckets.

**Additional context**
Maybe there is already an buildin way which might fit this feature but we didn't find it?

**Comment by klauspost** [comment, marked as answer]
See [bucket quota docs](https://docs.min.io/minio/baremetal/reference/minio-mc-admin/mc-admin-bucket-quota.html).

Since this sounds like a commercial setup look into our [subscription offers](https://min.io/pricing) where we can provide more detailed assistance.

**Reply by tsohst** [reply]
Hello @klauspost bucket quotas don't seem to help in this situation because it can still be possible to overload disk capacity on the infrastructure.

**Reply by klauspost** [reply]
@tsohst Correct. You will need to manage that.

**Reply by harshavardhana** [reply]
You want to limit your disk usage use DirectPV that allows for setting up disk quotas, as per your PVC requirements.
